File: hashmap/ransome-note/ransome-note.go
func canConstruct(ransomNote string, magazine string) bool {
    counts := make([]int, 26)

	for i := 0; i < len(magazine); i++{
		counts[magazine[i] - 'a']++
	}

	for i := 0; i < len(ransomNote); i++ {
		j := ransomNote[i] - 'a'
		counts[j]--
		if counts[j] < 0 {
			return false
		}
	}

	return true
}

File: hashmap/ransome-note/ransome-note.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    const counts = new Array(26).fill(0);

    for (let i = 0; i < magazine.length; i++) {
        counts[magazine.charCodeAt(i) - 97]++;
    }

    for (let i = 0; i < ransomNote.length; i++) {
        const j = ransomNote.charCodeAt(i) - 97;
        counts[j]--;
        if (counts[j] < 0) {
            return false;
        }
    }

    return true;
};